crates/sema/src/ty/ty.rs

```diff
@@ -424,6 +435,16 @@ impl<'gcx> Ty<'gcx> {
             return Ok(());
         }
 
+        // integers (signed and unsigned) can implicitly be converted to a larger integer. this also
+        // handles integer literals
+        if self.is_integer()
+            && other.is_integer()
+            && (!self.is_signed() || other.is_signed())
```

Comment:
?
```suggestion
            && self.is_signed() == other.is_signed()
```

Reply:
no, an unsigned integer can be assigned to a signed integer :)

Reply:
there should be a difference between literals which can coerce to either signed or unsigned, and typed integers, which can only upcast

Reply:
yeah makes sense